### ISSUE TITLE
xgboost: reloaded boosters error, or return raw logits labelled as probabilities

### ISSUE THREAD

**Issue by EmilHvitfeldt** [tidymodels/tidypredict#292]
`R/model-xgboost.R:72,124,239,384-397` discriminate old vs new xgboost API with `is.null(attr(model, "param"))`. That attribute is set by `xgb.train()` and **not** by `xgb.load()`, so a saved-and-reloaded booster takes the legacy branch and calls `xgb.model.dt.tree(text = ...)` — an argument xgboost 3.x has removed:

```
xgb loaded attr param NULL? TRUE   params$objective: NULL
Warning: Parameter(s) have been removed from this function: text.
ERROR: argument "model" is missing, with no default
```

Worse, when that branch is bypassed, `params$objective` is `NULL`, so `apply_xgb_objective()` (`:299-309`) only **warns** and returns the raw margin:

```
maxdiff vs predict (probabilities): 1.360145
got:      1.9261 1.9261 1.9261 -1.1018
expected: 0.7577 0.7577 0.7577  0.1315
```

The user gets logits presented as probabilities, behind a warning that misdescribes the cause ("If the objective is a custom function…").

**Fix.**

1. Stop using `attr(model, "param")` as the discriminator. `xgb.model.dt.tree(model = ...)` and `xgb.get.num.boosted.rounds()` work for both trained and loaded 3.x boosters; gate on the xgboost version instead.
2. Recover the objective from the saved JSON in `get_xgb_json_params()`, which already reads that file. The field is confirmed present — `objective":{"name":"binary:logistic","reg_loss_param":...` — so a regex alongside the existing `base_score` extraction is enough.

Note that fixing (1) must not un-mask a latent bug in the legacy path: `get_xgb_trees_df()` does `merge(trees, feature_names_tbl, by = "Feature")` at `:408` and **never restores the original row order**, unlike `get_xgb_trees_character()` which saves it at `:96` and restores at `:108-111`. `merge()` sorts by `Feature`, and `:414-420` then converts node IDs to row *positions* assuming row `i` is node `i-1`. Please fix both together.

Add an `xgb.save()` / `xgb.load()` round-trip test.
